### DISCUSSION TITLE
Out parameters for generic classes | Downcasting for generic

### DISCUSSION THREAD

**Discussion by RomanSoloweow** [dotnet/csharplang#5411] [Q&A]
Why is out not supported for parameters?

With the current implementation, we cannot do this:
```C#
class A {}
class B:A {}

class C<T> where T: A {}

C<B> t1 = new C<B>();

C<A> t2 = t1; //we cannot perform such a conversion
```

**Comment by HaloFour** [comment]
Generic variance does not extend to classes.  `B` may inherit from `A`, but `C<B>` does not inherit from `C<A>`.  If you could cast `List<B>` to `List<A>` you could be permitted to add values to it that are not `B` which would break type safety.

**Comment by fa1ntl1ght** [comment]
https://github.com/dotnet/csharplang/discussions/2498?
